maven-plugin/src/main/java/org/openmrs/maven/plugins/AddDependency.java

```diff
@@ -0,0 +1,134 @@
+package org.openmrs.maven.plugins;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.openmrs.maven.plugins.model.Artifact;
+import org.openmrs.maven.plugins.model.DistroProperties;
+
+import org.openmrs.maven.plugins.utility.NPMHelper;
+import org.openmrs.maven.plugins.utility.SDKConstants;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+
+@Mojo(name = "add-dependency", requiresProject = false)
+public class AddDependency extends AbstractTask {
+
+    /**
+     *  Path to the openmrs-distro.properties file to modify
+     */
+    @Parameter(property = "distro")
+    private String distro;
+
+    /**
+     * Type of the dependency to add
+     */
+    @Parameter(property = "type")
+    private String type;
```

Comment:
It would be good to document the acceptable values here so user's don't have to guess.